### DISCUSSION TITLE
How do I make a gedcom url?

### DISCUSSION THREAD

**Discussion by ItaiDanos** [PeWu/topola-viewer#69] [Q&A] [answered]
I want a way I can share the website to my family without them needing to use a file,
how can I do that?

Thanks

**Comment by PeWu** [comment, marked as answer]
One thing I've heard people do is upload the GEDCOM file to Google Drive and open the file from there.

1. Upload .GED file to [Google Drive](https://drive.google.com/)
2. Right-click on the uploaded file and select "Get link"
3. Change sharing settings to "Anyone with the link"
4. Click "Copy link" to copy the URL to clipboard
5. Open [Topola Viewer](https://pewu.github.io/topola-viewer/)
6. Click "Open URL"
7. Paste the copied URL
8. The genealogy tree should open
9. Copy the URL from the Topola Viewer browser tab and share it with your family

I hope this helps.

**Reply by ItaiDanos** [reply]
Thank you very much for the answer it works perfect :)

**Reply by mmcmonster** [reply]
Does this still work?  I tried with a viewable link from Google Drive as well as from Dropbox, and entering it into the "Load URL" dialog gives a "Failed to load file Forbidden" error message.

In particular, it will work when I access my host via http://localhost:3000/ but does not work when accessing via another computer on the local network or from the internet (with port forwarding of port 3000 to the server) ie: http://desktop.local:3000/

**Reply by PeWu** [reply]
Short answer: yes it works.

However, it gets complicated when you host Topola Viewer yourself.
This is an issue with [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing). To work around loading web resources from servers that restrict loading files with CORS Topola Viewer uses a [CORS-anywhere](https://github.com/Rob--W/cors-anywhere) proxy hosted at https://topola-cors.herokuapp.com/ This proxy only allows requests from the official Topola Viewer deployment and from localhost:3000 to allow development. Hosting Topola Viewer somewhere else means that if you want to load CORS-restricted URLs (like drive.google.com), you need to run your own CORS anywhere proxy. You can run it locally or online e.g. on Heroku.
